packages/components/src/components/post-user-menu/post-user-menu.tsx

```diff
@@ -0,0 +1,88 @@
+import { Component, Host, h, Prop, Watch, Element } from '@stencil/core';
+import { nanoid } from 'nanoid';
+import { version } from '@root/package.json';
+import { checkRequiredAndType } from '@/utils';
+
+export interface User {
+  name: string;
+  surname?: string;
+  email?: string;
+}
+
+@Component({
+  tag: 'post-user-menu',
+  styleUrl: 'post-user-menu.scss',
+  shadow: true,
+})
+export class PostUserMenu {
+  private readonly menuId = `p${nanoid(11)}`;
+
+  @Element() host: HTMLPostUserMenuElement;
+
+  /**
+   * A title for the user menu
+   */
+  @Prop() caption!: string;
+
+  @Watch('caption')
+  validateCaption() {
+    checkRequiredAndType(this, 'caption', 'string');
+  }
+
+  /**
+   * A descriptive text for the user avatar
+   */
+  @Prop() description!: string;
+
+  @Watch('description')
+  validateDescription() {
+    checkRequiredAndType(this, 'description', 'string');
+  }
+
+  /**
+   * An Object containing the personal data of the user currently logged-in.
+   */
+  @Prop() user?: User;
+
+  @Watch('user')
+  validateUser() {
+    checkRequiredAndType(this, 'user', 'object');
+  }
+
+  componentWillLoad() {
+    this.validateCaption();
+    this.validateDescription();
+    this.validateUser();
+  }
+
+  private renderAvatar() {
+    return (
+      <post-avatar
+        firstname={this.user.name}
+        lastname={this.user.surname}
+        email={this.user.email}
```

Comment:
The `post-avatar` also has the possibility to add a prop:
- `userid`
  This is currently not working, because the intranet team updated to M365.
  @gfellerph and I had once a talk with them, but without a final solution. I feel like now would be a good moment to discuss this with @gfellerph and the intranet team again, maybe as a follow up task of this one here.
- `imageSrc`
  Since the `email` property is tied to gravatar, this property could quickly become the preffered one for most of the projects that will use the user-menu. Therefore it would be good to implement it.